app/domain/references/routes.py

```diff
@@ -218,26 +221,48 @@ async def get_reference(
     return anti_corruption_service.reference_to_sdk(reference)
 
 
+def parse_identifiers(
+    identifier: Annotated[
+        list[str],
+        Query(
+            description=(
+                "A list of external identifier lookup strings "
+                "in the format `?({type}:?({other_type}:)){identifier}`. "
+                "These are accepted in multiple query parameters or as a CSV string."
+            ),
+            examples=[
+                "02e376ee-8374-4a8c-997f-9a813bc5b8f8",
+                "doi:10.1000/abc123",
+                "other:isbn:978-1-234-56789-0",
+            ],
+        ),
+    ],
+) -> list[IdentifierLookup]:
+    """Parse a list of identifier lookup strings from the query parameters."""
+    identifiers: list[IdentifierLookup] = []
+    for identifier_string in identifier:
+        if "," in identifier_string:
+            identifiers += parse_identifiers(identifier_string.split(","))
+        else:
+            identifiers.append(parse_identifier_lookup_from_string(identifier_string))
+    return identifiers
+
+
 @reference_router.get("/")
-async def get_reference_from_identifier(
-    identifier: str,
-    identifier_type: destiny_sdk.identifiers.ExternalIdentifierType,
+async def lookup_references(
     reference_service: Annotated[ReferenceService, Depends(reference_service)],
     anti_corruption_service: Annotated[
         ReferenceAntiCorruptionService, Depends(reference_anti_corruption_service)
     ],
-    other_identifier_name: str | None = None,
-) -> destiny_sdk.references.Reference:
-    """Get a reference given an external identifier."""
-    external_identifier = ExternalIdentifierSearch(
-        identifier=identifier,
-        identifier_type=identifier_type,
-        other_identifier_name=other_identifier_name,
-    )
-    reference = await reference_service.get_reference_from_identifier(
-        external_identifier
-    )
-    return anti_corruption_service.reference_to_sdk(reference)
+    identifiers: Annotated[list[IdentifierLookup], Depends(parse_identifiers)],
```

Comment:
I wonder if this is where we might do some limiting to prevent potential DoSes.

```suggestion
    identifiers: Annotated[list[IdentifierLookup], Len(max_length=settings.max_identifier_lookup_count), Depends(parse_identifiers)],
```

Reply:
Actioned this and also a hard cap in the validation (to validate against comma-separated identifiers).